backend/mqttpubsub/backend.go

```diff
@@ -33,7 +36,14 @@ func NewBackend(server, username, password string) (*Backend, error) {
 	opts.SetPassword(password)
 	opts.SetOnConnectHandler(b.onConnected)
 	opts.SetConnectionLostHandler(b.onConnectionLost)
-
+	
+	if len(cafile) != 0 {
+		tlsconfig, err := NewTLSConfig(cafile)
+		if(err == nil) {
+			opts.SetClientID("ssl-client").SetTLSConfig(tlsconfig)
```

Comment:
@brocaar, you are perfectly right on this.

However just for let you know some MQTT server allow load balacing based on the clientID, it is quite a nice feature.

https://vernemq.com/docs/configuration/balancing.html